dimelo/parse_bam.py

```diff
@@ -552,42 +564,61 @@ def check_bam_format(
-        motif, pos = basemod.split(",")
-        base = motif[int(pos)]
-        basemods_found_dict[base] = False
+    mod_codes_dict = {}
+    mod_codes_found_dict = defaultdict(set)
+    for motif in motifs:
+        parsed_motif = utils.ParsedMotif(motif)
+        mod_codes_dict[parsed_motif.modified_base] = parsed_motif.mod_codes
+        basemods_found_dict[parsed_motif.modified_base] = False
 
     input_bam = pysam.AlignmentFile(bam_file)
 
     try:
         for counter, read in enumerate(input_bam.fetch()):
             read_dict = read.to_dict()
             for tag_string in read_dict["tags"]:
-                tag_fields = tag_string.split(",")[0].split(":")
-                tag = tag_fields[0]
-                # tag_type = tag_fields[1]
-                tag_value = tag_fields[2]
+                tag = tag_string.split(",")[0].split(":")[0]
                 if tag == "Mm" or tag == "Ml":
                     raise ValueError(
                         f'Base modification tags are out of spec (Mm and Ml instead of MM and ML). \n\nConsider using "modkit update-tags {str(bam_file)} new_file.bam" in the command line with your conda environment active and then trying with the new file. For megalodon basecalling/modcalling, you may also need to pass "--mode ambiguous.\nBe sure to index the resulting .bam file."'
                     )
                 elif tag == "MM":
-                    if (
-                        len(tag_value) > 0
-                        and tag_value[-1] != "?"
-                        and tag_value[-1] != "."
-                    ):
-                        raise ValueError(
-                            f'Base modification tags are out of spec. Need ? or . in TAG:TYPE:VALUE for MM tag, else modified probability is considered to be implicit. \n\nConsider using "modkit update-tags {str(bam_file)} new_file.bam --mode ambiguous" in the command line with your conda environment active and then trying with the new file.'
-                        )
-                    else:
-                        if len(tag_value) > 0 and tag_value[0] in basemods_found_dict:
-                            if tag_value[2] in utils.BASEMOD_NAMES_DICT[tag_value[0]]:
-                                basemods_found_dict[tag_value[0]] = True
-                            else:
-                                raise ValueError(
-                                    f'Base modification name unexpected: {tag_value[2]} to modify {tag_value[0]}, should be in set {utils.BASEMOD_NAMES_DICT[tag_value[0]]}. \n\nIf you know what your mod names correspond to in terms of the latest .bam standard, consider using "modkit adjust-mods {str(bam_file)} new_file.bam --convert 5mC_name m --convert N6mA_name a --convert other_basemod_name correct_label" and then trying with the new file. Note: currently supported mod names are {utils.BASEMOD_NAMES_DICT}'
-                                )
+                    for tag_substring in tag_string.split(";"):
```

Comment:
I am choosing to believe that you have done things correctly here and moving on. Just don't think it's worth it for me to dig into it, unless there's something you think I need to review here.

Reply:
Let's just run a bunch of bam files from different sources and see; this has currently only been tested on Megalodon and r10 Dorado files